execute/executor.go

```diff
@@ -330,15 +333,28 @@ func (v *createExecutionNodeVisitor) generateResult(resultName string, node plan
 
 // getResultName will offer a "best guess" name for a given node's result.
 //
-// For nodes that have side-effects or happen to be a Parallel Merge, the result
-// will be based on the node ID. For other cases, the default yield name of
-// `_result` is used.
-func getResultName(node plan.Node, spec plan.ProcedureSpec, isParallelMerge bool) string {
+// For nodes that have side-effects, the result will be based on the node ID.
+// If the node is a parallel merge node, then it was added by the planner; in
+// this case check the side-effect status of the node's predecessor. For other
+// cases, the default yield name of `_result` is used.
+func getResultName(node plan.Node, spec plan.ProcedureSpec, isParallelMerge bool) (string, error) {
 	name := plan.DefaultYieldName
-	if plan.HasSideEffect(spec) || isParallelMerge {
+
+	// Parallel merge nodes are added by the planner and can mask the presence
+	// of a side-effect result; skip them.
+	if isParallelMerge {
+		if len( node.Predecessors() ) != 1 {
+			return "", errors.New(codes.Internal, "parallel merge must have a single predecessor")
```

Comment:
I think you mentioned possibly being able to enforce this during planning. The check here should probably be here regardless, but I'm wondering if there'll be a follow-on to this follow-on.

Reply:
Yes, was planning on adding to the planner attributes a way to say that the node with the attribute must have at least one successor, or that it must have at least one predecessor.